### PR TITLE
Make StatSlot.OnCompleted sense

### PR DESCRIPTION
**Pull request by louyuting**
### Describe what this PR does / why we need it
Make StatSlot.OnCompleted sense

### Describe how to verify it
Ut 

### Special notes for reviews

### CODE REVIEW

File: core/base/slot_chain.go [hunk 1]
```diff
@@ -185,7 +185,14 @@ func (sc *SlotChain) Entry(ctx *EntryContext) *TokenResult {
 }
 
 func (sc *SlotChain) exit(ctx *EntryContext) {
+	if ctx == nil || ctx.Entry() == nil {
+		return
+	}
 	for _, s := range sc.stats {
+		// The OnCompleted is called only when entry passed
+		if ctx.IsBlocked() {
```
Comment (sczyh30) [inline core/base/slot_chain.go:193]: Maybe just checking it outside is better?

Comment (louyuting) [reply core/base/slot_chain.go:193]: make sense

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Thanks for contributing!

**Pull request merged**